components/kyma-environment-broker/cmd/broker/main.go

```diff
@@ -55,18 +72,57 @@ func main() {
 		provisionerClient = provisioner.NewProvisionerClient(cfg.Provisioning.URL, true)
 	}
 
+	secrets, err := newSecretsInterface(cfg.Director.Namespace)
+	fatalOnError(err)
+	oauthClient := oauth.NewOauthClient(newHTTPClient(false), secrets, cfg.Director.OauthCredentialsSecretName)
+	fatalOnError(oauthClient.WaitForCredentials())
+	tkn, err := oauthClient.GetAuthorizationToken()
+	fatalOnError(err)
+	fmt.Println("DUPA:", tkn.AccessToken)
+	director.NewDirectorClient(oauthClient)
+
 	db, err := storage.New(cfg.Database.ConnectionURL())
 	fatalOnError(err)
 
-	broker, err := broker.NewBroker(provisionerClient, cfg.Provisioning, db.Instances())
+	kymaEnvBroker, err := broker.NewBroker(provisionerClient, cfg.Provisioning, db.Instances())
 	fatalOnError(err)
 
-	brokerAPI := brokerapi.New(broker, logger, brokerCredentials)
+	brokerAPI := brokerapi.New(kymaEnvBroker, logger, brokerCredentials)
 	r := handlers.LoggingHandler(os.Stdout, brokerAPI)
 
 	fatalOnError(http.ListenAndServe(cfg.Host+":"+cfg.Port, r))
 }
 
+func newHTTPClient(skipCertVeryfication bool) *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: skipCertVeryfication},
+		},
+		Timeout: 30 * time.Second,
+	}
+}
+
+func newSecretsInterface(namespace string) (v1.SecretInterface, error) {
```

Comment:
I am not convinced about `interface` part in the function name :)

Reply:
ok will change to client